wolfcrypt/src/port/kcapi/kcapi_ecc.c

```diff
@@ -190,14 +192,52 @@ int KcapiEcc_Sign(ecc_key* key, const byte* hash, word32 hashLen, byte* sig,
         }
     }
     if (ret == 0) {
-        ret = kcapi_akcipher_sign(key->handle, hash, hashLen, sig, *sigLen,
+        if ((long int)sig % sysconf(_SC_PAGESIZE) != 0) {
+            ret = posix_memalign((void*)&sig_aligned,
```

Comment:
Have you considered allocating `sysconf(_SC_PAGESIZE) * 2`? If either is not aligned then just allocate two pages and use offset of page for each?

Reply:
Makes sense. Done.